### PR TITLE
Fix support for parsing of classes located in runtime jars

### PR DESCRIPTION
**Pull request by stuhood**

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

**Comment by eed3si9n** [comment]
Fixes #2263?

**Comment by eed3si9n** [comment]
Coincidently I haven't merged months old sbt/io/pull/13 yet. Could you update that PR too?

File: util/io/src/main/scala/sbt/IO.scala [hunk 1]
```diff
@@ -36,39 +36,43 @@ object IO {
...
-      // NB: This assumes that classes without code sources are System classes, and thus located in
-      // jars. It assumes that `urlAsFile` will truncate to the containing jar file.
-      val clsfile = s"${cl.getName.replace('.', '/')}.class"
-      Option(ClassLoader.getSystemClassLoader.getResource(clsfile))
-        .flatMap {
-          urlAsFile
-        }.getOrElse {
-          sys.error("No class location for " + cl)
-        }.toURI.toURL
+  def classfileLocation(cl: Class[_]): URL = {
```
Comment (dwijnand) [inline util/io/src/main/scala/sbt/IO.scala:42]: Bincompat?

**Comment by dwijnand** [comment]
Is this a fix to an improvement for which we have notes or is it a fix that requires its own notes?

**Comment by eed3si9n** [comment]
@dwijnand I think this is an improvement to #2214, which is a part of #2085, which does have a note entry.

**Comment by dwijnand** [comment]
Cool thanks.

File: util/io/src/main/scala/sbt/IO.scala (hunk 1 above)
Comment (dwijnand) [inline util/io/src/main/scala/sbt/IO.scala:42]: @eed3si9n Do we want to do something about this binary compatibility breakage before cutting a 0.13.10 release?

Comment (eed3si9n) [inline util/io/src/main/scala/sbt/IO.scala:42]: Yes. Could you send a PR or issue against 0.13 to fix this?

Comment (dwijnand) [inline util/io/src/main/scala/sbt/IO.scala:42]: sure

**Comment by dwijnand** [comment]
Also, this hasn't been forward-ported to 1.0, which I realise is a little hard as it's touching files in different modules..

**Pull request merged**